lib/liquid/condition.rb

```diff
@@ -47,6 +48,11 @@ def evaluate(context = Context.new)
       when :or
         result || @child_condition.evaluate(context)
       when :and
+        @@depth += 1
+        if @@depth >= 500
```

Comment:
This should be a class variable so the library user has some control over the value.